src/core/include/iDynTree/Core/CubicSpline.h

```diff
@@ -36,13 +36,17 @@ namespace iDynTree
         
     public:
         CubicSpline();
+
+        CubicSpline(unsigned int buffersDimension);
+
         ~CubicSpline();
```

Comment:
In general, we should try to respect the [rule of 5](http://en.cppreference.com/w/cpp/language/rule_of_three) in our code. In this case in particular, I think we should either remove the definition of this destructor (if it is not used) or specifically mark it as `default` using `= default` (see http://en.cppreference.com/w/cpp/language/destructor).